src/backend/joanie/core/api/admin.py

```diff
@@ -308,3 +308,50 @@ class ContractDefinitionViewSet(viewsets.ModelViewSet):
     permission_classes = [permissions.IsAdminUser & permissions.DjangoModelPermissions]
     serializer_class = serializers.AdminContractDefinitionSerializer
     queryset = models.ContractDefinition.objects.all()
+
+
+class OrderGroupViewSet(
+    mixins.CreateModelMixin,
+    mixins.UpdateModelMixin,
+    mixins.RetrieveModelMixin,
+    mixins.ListModelMixin,
+    viewsets.GenericViewSet,
+):
+    """
+    OrderGroup ViewSet
+    """
+
+    authentication_classes = [SessionAuthenticationWithAuthenticateHeader]
+    permission_classes = [permissions.IsAdminUser & permissions.DjangoModelPermissions]
+    serializer_class = serializers.AdminOrderGroupSerializer
+    queryset = models.OrderGroup.objects.all().select_related("product")
+    ordering = "created_on"
+
+    def get_queryset(self):
+        product_id = self.kwargs.get("product_id")
+        return super().get_queryset().filter(product__id=product_id)
+
+    def create(self, request, *args, **kwargs):
+        data = request.data
+        data["product"] = kwargs.get("product_id")
+        return super().create(request, *args, **kwargs)
+
+    def update(self, request, *args, **kwargs):
+        data = request.data
+        data["product"] = kwargs.get("product_id")
+        return super().update(request, *args, **kwargs)
+
+    def partial_update(self, request, *args, **kwargs):
+        data = request.data
+        data["product"] = kwargs.get("product_id")
+        return super().partial_update(request, *args, **kwargs)
+
+    def list(self, request, *args, **kwargs):
+        data = request.data
+        data["product"] = kwargs.get("product_id")
+        return super().list(request, *args, **kwargs)
+
+    def retrieve(self, request, *args, **kwargs):
+        data = request.data
+        data["product"] = kwargs.get("product_id")
+        return super().retrieve(request, *args, **kwargs)
```

Comment:
I think it's useless to override `list`, `retrieve` and `partial_update`. Indeed they are all using `get_queryset` method to retrieve resource(s) under the hood.